components/postcode-lookup/index.js

```diff
@@ -0,0 +1,339 @@
+          defaults.ADDRESS_LOOKUP_SEARCH_TITLE,
+        addressLookupTitle: conditionalTranslate('pages.address-lookup.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_TITLE,
+        addressDetailsTitle: conditionalTranslate('pages.address-details.title', req.translate) ||
+          defaults.ADDRESS_DETAILS_TITLE,
+        lookupProblemTitle: conditionalTranslate('pages.address-lookup-problem.title', req.translate) ||
+          defaults.ADDRESS_LOOKUP_PROBLEM_TITLE,
+        route: this.options.route,
+        editLink,
+        cantFind,
+        searchByPostcodeLink,
+        postcodeError,
+        section
+      });
+    }
+
+    postcode(req, res, callback) {
+      // Clear the value stored in the addresses radio button group
+      req.sessionModel.set(`${addressKey}-select`, '');
+      // Call OS Places API to return list of addresses by postcode
+      const enteredPostcode = req.form.values[`${addressKey}-postcode`];
+
+      axios.get(apiURL + '?postcode=' +  enteredPostcode + '&key=' + apiKey)
+        .then(function (response) {
+          const addressList = [];
+          _.forOwn(response.data.results, function (value) {
+            addressList.push(value.DPA);
+          });
+
+          req.sessionModel.set(`${addressKey}-addresses`, addressList);
+          return callback();
+        }).catch(error => {
+          if(error.response.status === '404' || error.response.status === '429' ||
+            error.response.status === '503' || error.response.status === '500') {
+            req.query.step === 'postcodeSearchIssue';
+          }
+          return callback(error);
+        });
+    }
+
+    setupRadioButton(req, addresses) {
+      req.form.options.fields[`${addressKey}-addresses`] = {
+        mixin: 'radio-group',
+        isPageHeading: true,
+        validate: ['required'],
+        options: addresses.map(obj => {
+          const addressValue = obj.ADDRESS;
+          return {
+            label: addressValue,
+            value: addressValue
+          };
+        })
+      };
+    }
+
+    saveValues(req, res, callback) {
+      const step = req.query.step;
+      if (step === 'postcode') {
+        postcode = req.form.values[`${addressKey}-postcode`];
+        req.form.values[`${addressKey}`] = '';
+        req.form.values[`${addressKey}-address-line-2`] = '';
+        req.form.values[`${addressKey}-town-or-city`] = '';
+        req.form.values[`${addressKey}-postcode-manual`] = '';
+        return this.postcode(req, res, err => {
+          if (err) {
+            req.sessionModel.set('addresses', []);
+            return res.redirect(req.baseUrl + '/postcode?step=postcodeSearchIssue');
+          }
+          return super.saveValues(req, res, callback);
+        });
+      } else if (step === 'lookup') {
+        const addressResult = req.form.options.fields[`${addressKey}-select`].options.filter(obj => {
```

Comment:
Should do some sanitisation of the input values gathered, strip out dodgy characters like < , {, ! and ?

Reply:
Added logic to remove any instances of those characters